docs/guidelines/git-signed-commits.md

```diff
@@ -0,0 +1,172 @@
+
+### Adding a SSH key for signing
+
+!!! Note Use multiple SSH keys
+
+    We recommend using different SSH keys for authentication and signing. This may add some extra complexity and it may provide a more robust set-up with looser coupling of key components of the SDLC.
+
+To create a new SSH key for signing you can use the following command (alter the date manually):
+
+```shell
+ssh-keygen -t ed25519 -f ~/.ssh/git_ssh_signing_key_1 -C "Created on <date>, for larskaare on github.com"
+```
+
+This will create a SSH signing key and add a comment on date and purpose. Add a passphrase to the key. Success full key generation will output the key fingerprint and a randomart image (randomart is supposed to be a visualisation making it easier to validate keys - and identify changes)
+
+Add the new key to the ssh-agent
+
+```shell
+ssh-add ~/.ssh/git_ssh_signing_key_1
+```
+
+### [Configure Git](https://docs.github.com/en/authentication/managing-commit-signature-verification/telling-git-about-your-signing-key#telling-git-about-your-ssh-key) to use the SSH key
+
+We will configure the git global settings to use the new SSH key for signing commits locally. Examples assume you created the key as defined above.
+
+```shell
+git config --global gpg.format ssh
+git config --global user.signingkey $HOME/.ssh/git_ssh_signing_key_1.pub
+git config --global commit.gpgsign true
+```
+
+These lines will tell git to use SSH for signing commit, tell git where to find the key that should be used and then tell git to always sign commits. If you do not add this last line you will specifically have to add the `-S` parameter for each commit you can to sign.
+
+### Examining the git log
+
+To verify that commits are signed locally you can use the following command:
+
+```shell
+git log --show-signature
+```
+
+When you run this command on a newly configured system you may get an error message like `error: gpg.ssh.allowedSignersFile needs to be configured and exist for ssh signature verification` In order for git to verify signatures locally you need to add the public keys that are used to sign to a file that Git will use.
+
+We will create the allowed_signers file. It typically has the format like "signer email" "key-type" "key-body". The key in question is the public key of the SSH key we use to sign our commits.
+
+```shell
+git config --global gpg.ssh.allowedSignersFile $HOME/.ssh/allowed_signers
+echo $(git config --get user.email) \
+     $(cat ~/.ssh/git_ssh_signing_key_1.pub) \
+     | awk '{print $1,$2,$3}' >> ~/.ssh/allowed_signers
+```
+
+- The first line tells Git which file we are going to use for allowed_signers
+- The second line emits the information on our key and adds it to the allowed_signers file. The file is created if it does not exist
+
+When this is done you can view the git log and verify the signature.
+
+```shell
+git log --show-signature
```

Comment:
This now shows  that _my_ commits are being properly signed, but for commits by other users (in this repository) I get the error
```
gpg: Can't check signature: No public key
```

Reply:
As expected. I'll add a tip later on adding other public keys to the allowed_signers file